### ISSUE TITLE
No selection box outline visible when pointing node behind liquid

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#5780]
If you point a node behind a liquid, but are not inside a liquid yourself, the selection box of this node is not visible. This bug is very obvious in shallow water:

![No selection box visible](https://i.imgur.com/iu9stmX.png)

Note the lack of a visible selection box. The pointed node here is the dirt. It is definitely in pointing range, I can punch and mine the pointed dirt.
Note that you have to be outside of the water for this bug to happen. If you are inside the water and point the dirt again, the selection box is visible.

Happens in 0.4.15 and later.

**Comment by bigfoot547** [comment]
Particles also don't show up in water.

**Comment by nerzhul** [comment]
yeah, it's because you point at the liquid not the node behind not sure if a fix should be done for this @paramat ?

**Comment by paramat** [comment]
I guess the selected node is always the first non-air node, 'liquids pointable = true' https://github.com/minetest/minetest_game/blob/master/mods/flowers/init.lua#L264 then makes the selectionbox appear for the pointed water node. I don't think it's a bug, just something to consider.

**Comment by Wuzzy2** [comment]
You both failed to understand the bug. I try again, but this time in pictures.

Pointing with the **hand** (!) (`liquids_pointable=false`) on a solid node behind a liquid fails to render any seletion box whatsoever, **neither on the water nor on the solid node** (buggy behaviour):
![](https://i.imgur.com/7xrIwXO.png)

Comparison with bucket (correct behaviour):
![](https://i.imgur.com/MbUxfX9.png)

Proof that I really pointed the sand node, and not just the liquid above it:
![](https://i.imgur.com/DKw9K4E.png)

Here the same node is pointed, the only difference is that I'm underwater (correct behaviour):
![](https://i.imgur.com/bd2qxoC.png)

How to reproduce:

* Stand above shallow water as shown in screenshot 1
* Point into the sand or any other solid node under the water, while being outside of the water
* Expect a selection box to be rendered but don't see any selection box, neither on water, nor on the sand

Tell me again that this is not a bug.

I think the only situation where there should be no selection box rendered whatsoever is when you actually point nothing (you point into air or your target is out of range). The fact that no selection box is rendered in screenshot 1 (falsely) suggests that you don't point anything or the sand is out of range.

**Comment by paramat** [comment]
Yeah i got this wrong.

**Comment by DS-Minetest** [comment]
>For comparison, here's the same situation with bucket (liquids_pointable=true) (**correct behaviour**):<details><summary>image</summary>![image](https://camo.githubusercontent.com/4ad1e5eb15b97340b06de807e8da092446120418/68747470733a2f2f692e696d6775722e636f6d2f4d6255786658392e706e67)</details>

Just a question, is it really correct that the selection lines are not visible under the water surface. I see a square, no cube. I will zoom to look if/where it's cut off.

**Comment by Wuzzy2** [comment]
Oh. Right. Nice catch.
Actually the selection box should be the full box, not just the surface of it.
Apparently the lines become invisible as soon they go into the liquid.
Try throwing an item into low flowing water, then point it.

**Comment by DS-Minetest** [comment]
Yep, same with items:
![screenshot_20170531_153037](https://cloud.githubusercontent.com/assets/7613443/26634244/4869f108-4616-11e7-8324-cbe1b9314840.png)
